apps/explorer/app/tx/[id]/opengraph-image.tsx

```diff
@@ -14,65 +28,50 @@ export const size = {
 
 export const contentType = 'image/png'
 
+const NotFoundImage = (id: string) =>
+  getOGImage(
+    {
+      id,
+      title: truncate(id, 30),
+      subtitle: 'transaction',
+      initials: 'T',
+    },
+    size
+  )
+
 export default async function Image({ params }) {
   const id = params?.id as string
+  let searchResult: SearchResultType
+  let transaction: ExplorerTransaction | ExplorerV2Transaction
+  let transactionChainIndices: ChainIndex | ChainIndex[]
 
-  const [
-    [transaction, transactionError],
-    [transactionChainIndices, transactionChainIndicesError],
-    [currentTip, currentTipError],
-  ] = await Promise.all([
-    to(
-      getExplored().transactionByID({
-        params: {
-          id,
-        },
-      })
-    ),
-    to(
-      getExplored().transactionChainIndices({
-        params: {
-          id,
-        },
-      })
-    ),
-    to(getExplored().consensusTip()),
-  ])
+  // Do we have a v1 or v2 transaction?
+  try {
+    searchResult = await fetchSearchType(id)
```

Comment:
Can move all logic in here to simplify reassignment, can reduce to a single catch at the bottom.